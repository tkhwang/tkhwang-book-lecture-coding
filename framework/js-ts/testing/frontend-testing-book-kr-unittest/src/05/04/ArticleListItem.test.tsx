import { render, screen } from "@testing-library/react";
import { ArticleListItem, ItemProps } from "./ArticleListItem";

const item: ItemProps = {
  id: "howto-testing-with-typescript",
  title: "타입스크립트를 사용한 테스트 작성법",
  body: "테스트 작성 시 타입스크립트를 사용하면 테스트의 유지 보수가 쉬워진다",
};

test("링크에 id로 만든 URL을 표시한다", () => {
  render(<ArticleListItem {...item} />);
  expect(screen.getByRole("link", { name: "더 알아보기" })).toHaveAttribute(
    "href",
    "/articles/howto-testing-with-typescript"
  );
});

test("Snapshot: 개별 아이템을 표시한다", () => {
  const { container } = render(<ArticleListItem {...item} />);
  expect(container).toMatchSnapshot();
});
