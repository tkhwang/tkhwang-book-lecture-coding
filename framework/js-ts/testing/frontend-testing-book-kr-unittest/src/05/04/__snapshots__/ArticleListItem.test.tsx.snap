// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot: 개별 아이템을 표시한다 1`] = `
<div>
  <li>
    <h3>
      타입스크립트를 사용한 테스트 작성법
    </h3>
    <p>
      테스트 작성 시 타입스크립트를 사용하면 테스트의 유지 보수가 쉬워진다
    </p>
    <a
      href="/articles/howto-testing-with-typescript"
    >
      더 알아보기
    </a>
  </li>
</div>
`;
